apps/server/src/shared/domain/entity/lesson.entity.ts

```diff
@@ -9,6 +9,19 @@ export interface ILessonProperties {
 	hidden?: boolean;
 	course: Course;
 	position?: number;
+	contents: IComponentProperties[] | [];
+}
+
+export interface IContentProperties {
+	content: {
+		text: string;
+	};
+}
+
+export interface IComponentProperties extends IContentProperties {
+	title: string;
+	hidden?: boolean;
+	component: string;
```

Comment:
we could define an enum with all possible types (TODO - maybe check in prod db, these types are the only ones possible - even if we have buttons for them only, maybe there is old data?)
```
enum ComponentType {
	TEXT = 'text',
	INTERNAL = 'internal',
	LERNSTORE = 'resouces',
	GEOGEBRA = 'geoGebra',
	EHTERPAD = 'Ehterpad',
}
```

and have 
```
component: ComponentType
```

Reply:
we can try something like this

```
export enum ContentComponent {
	TEXT = 'text',
	TASK = 'internal',
	LERNSTORE = 'resouces',
	GEOGEBRA = 'geoGebra',
	EHTERPAD = 'Ehterpad',
}

export interface IComponentTextProperties {
	text: string;
}

export interface IComponentTaskProperties {
	url: string;
}
....

export interface IComponentProperties {
	title: string;
	hidden?: boolean;
	component: ContentComponent;
	user: User;
	content: IComponentTextProperties | IComponentTaskProperties;
}

```